web/pdf_page_view.js

```diff
@@ -1240,4 +1277,229 @@ class PDFPageView extends PDFPageViewBase {
+    const minY = Math.max(0, visibleArea.minY - overflowHeight);
+    const maxY = Math.min(viewport.height, visibleArea.maxY + overflowHeight);
+    const width = maxX - minX;
+    const height = maxY - minY;
+
+    this.detailArea = { minX, minY, width, height };
+
+    this.reset({ keepCanvas: true });
+  }
+
+  async draw() {
+    const initialRenderingState = this.renderingState;
+    if (initialRenderingState !== RenderingStates.INITIAL) {
+      console.error("Must be in new state before drawing");
+      this.reset(); // Ensure that we reset all state to prevent issues.
+    }
+    const { div, pdfPage, viewport } = this.pageView;
+
+    if (!pdfPage) {
+      this.renderingState = RenderingStates.FINISHED;
+      throw new Error("pdfPage is not loaded");
+    }
+
+    this.renderingState = RenderingStates.RUNNING;
+
+    const canvasWrapper = this.pageView._ensureCanvasWrapper();
+
+    const { canvas, prevCanvas, ctx } = this._createCanvas(
+      newCanvas => {
+        // If there is already the background canvas, inject this new canvas
+        // after it. We cannot simply use .append because all canvases must
+        // be before the SVG elements used for drawings.
+        if (canvasWrapper.firstElementChild?.tagName === "CANVAS") {
+          canvasWrapper.firstElementChild.after(newCanvas);
+        } else {
+          canvasWrapper.prepend(newCanvas);
+        }
+      },
+      // If there is already the lower resolution canvas behind,
+      // we don't show the new one until when it's fully ready.
+      this.pageView.renderingState === RenderingStates.FINISHED ||
+        initialRenderingState === RenderingStates.FINISHED
+    );
+    const { width, height } = viewport;
+
+    const area = this.detailArea;
+
+    const { devicePixelRatio = 1 } = window;
+    const transform =
+      devicePixelRatio !== 1
+        ? [
+            devicePixelRatio,
+            0,
+            0,
+            devicePixelRatio,
+            -area.minX * devicePixelRatio,
+            -area.minY * devicePixelRatio,
+          ]
+        : null;
+
+    canvas.width = area.width * devicePixelRatio;
+    canvas.height = area.height * devicePixelRatio;
+    canvas.style.position = "absolute";
+    canvas.style.display = "block";
+    canvas.style.width = `${(area.width * 100) / width}%`;
+    canvas.style.height = `${(area.height * 100) / height}%`;
+    canvas.style.top = `${(area.minY * 100) / height}%`;
+    canvas.style.left = `${(area.minX * 100) / width}%`;
+
+    const renderingPromise = this._drawCanvas(
+      {
+        canvasContext: ctx,
+        transform,
+        viewport,
+        pageColors: this.pageColors,
+      },
+      prevCanvas,
+      () => {
+        this.eventBus.dispatch("pagerendered", {
+          source: this,
+          pageNumber: this.id,
+          cssTransform: false,
+          timestamp: performance.now(),
+          error: this._renderError,
+        });
```

Comment:
Maybe we should introduce helper-methods in the base-class for dispatching this event, to reduce duplication?